exist-core/src/main/java/org/exist/xquery/FunctionTypeCheck.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.xquery;
+
+import org.exist.dom.persistent.DocumentSet;
+import org.exist.xquery.util.ExpressionDumper;
+import org.exist.xquery.value.*;
+
+/**
+ * Check a function parameter type at runtime.
+ *
+ * @author wolf
+ */
+public class FunctionTypeCheck extends AbstractExpression {
+
+    private final Expression expression;
+    private final FunctionParameterFunctionSequenceType requiredType;
+
+    public FunctionTypeCheck(XQueryContext context, final FunctionParameterFunctionSequenceType requiredType, Expression expr) {
+        super(context);
+        this.requiredType = requiredType;
+        this.expression = expr;
+    }
+
+    /* (non-Javadoc)
+     * @see org.exist.xquery.Expression#analyze(org.exist.xquery.AnalyzeContextInfo)
+     */
+    public void analyze(AnalyzeContextInfo contextInfo) throws XPathException {
```

Comment:
Please remove the `/* (non-Javadoc)` (which is not used since very old versions of auto-code generation in Eclipse IDE), and replace with JavaDoc. This applies throughout this class...